ebs/error_budget.py

```diff
@@ -417,6 +410,409 @@ def run_etc(self, wfe, wfsc_factor, sensitivity
+                        self.write_csv(var_name)
+                        self.exosims_pars_dict['starlightSuppressionSystems']\
+                                [0]['core_contrast'] = self.contrast_filename
+                    self.write_ppFact_fits()
+                    self.exosims_pars_dict['ppFact'] = self.ppFact_filename
+                if var_name == 'throughput':
+                    self.write_csv(var_name)
+                    self.exosims_pars_dict['starlightSuppressionSystems']\
+                            [0]['core_thruput'] = self.throughput_filename
+            else:
+                print(key+" not found in attributes list")
+
+    def log_prior(self, values):
+        joint_prob = 0.0
+        counter = 0
+        for i, var_name in enumerate(self.config['mcmc']['variables'].keys()):
+            prior_ftn = np.array(
+                    self.config['mcmc']['variables'][var_name]['prior_ftn']
+                                )
+            index = np.where(prior_ftn!='nan')
+            all_args = [np.array(item) for item in 
+                    self.config['mcmc']['variables'][var_name]['prior_args']
+                        .values()]
+            for m, row_index in enumerate(index[0]):
+                if len(index) > 1:
+                    col_index = index[1][m]
+                    ftn = getattr(pdf, prior_ftn[row_index, col_index])
+                    args = [page[row_index][col_index] for page in all_args]
+                else:
+                    ftn = getattr(pdf, prior_ftn[row_index])
+                    args = [page[row_index] for page in all_args]
+                prob = ftn(values[counter], *args)
+                counter += 1
+                if np.isinf(prob):
+                    return prob
+                else: 
+                    joint_prob += prob
+        return joint_prob 
+
+    def log_merit(self, values):
+        self.update_attributes(values)
+        int_time = self.run_exosims()[0]
+        if np.isnan(int_time.value).any():
+            return -np.inf
+        else:
+            ftn_name = self.config['mcmc']['merit']['ftn']
+            args = [arg for arg in self.config['mcmc']['merit']['args']\
+                    .values()]
+            ftn = getattr(pdf, ftn_name)
+            mean_int_time = np.array(int_time.value).mean() 
+            return ftn(mean_int_time, *args)
+
+#    def log_probability(self, values):
+#        log_prior = self.log_prior(values)
+#        if np.isinf(log_prior):
+#            return -np.inf
+#        log_merit = self.log_merit(values)
+#        if np.isinf(log_merit):
+#            return -np.inf
+#        log_probability = log_prior + log_merit
+#        return log_probability
+
+    def run_mcmc(self, parallel=False):
+        self.initialize_for_exosims()
+        pos = self.initialize_walkers()
+        nwalkers, ndim = pos.shape
+        backend = emcee.backends.HDFBackend(self.config['mcmc']['backend_path'])
```

Comment:
is the HDF5 backend file just a blank HDF5 file? If so there should be a line in here like 

```suggestion
        if not os.path.exists(self.config['mcmc']['backend_path']):
            file = h5py.File("example.hdf", "r")
            file.close()
            self.config['mcmc']['backend_path'] = 'example.hdf'
        backend = emcee.backends.HDFBackend(self.config['mcmc']['backend_path'])
```
that will generate the hdf file if it doest already exist. If that file needs to contain information and is not blank, an example should be added to the repo.